apps/nerves_hub_www/lib/nerves_hub_www_web/live/device_live/index.ex

```diff
@@ -148,4 +163,33 @@ defmodule NervesHubWWWWeb.DeviceLive.Index do
       end
     end
   end
+
+  defp do_reboot(socket, :allowed, device) do
+    AuditLogs.audit!(socket.assigns.user, device, :update, %{reboot: true})
+
+    socket.endpoint.broadcast_from(self(), "device:#{device.id}", "reboot", %{})
+
+    socket =
+      socket
+      |> put_flash(:info, "Device Reboot Requested")
+      |> assign(:device, %{device | status: "reboot-requested"})
+
+    {:noreply, socket}
+  end
+
+  defp do_reboot(socket, :blocked) do
+    msg = "User not authorized to reboot this device"
+
+    AuditLogs.audit!(socket.assigns.user, socket.assigns.device, :update, %{
+      reboot: false,
+      message: msg
+    })
+
+    socket =
+      socket
+      |> put_flash(:error, msg)
+      |> assign(:device, %{socket.assigns.device | status: "reboot-blocked"})
+
+    {:noreply, socket}
+  end
```

Comment:
I think we should probably add a test for this. I can do that, if you like. The example I would start with is [here](https://github.com/verypossible/nerves_hub_web/blob/main/apps/nerves_hub_www/test/nerves_hub_www_web/live/device_live_show_test.exs#L21).